pyccl/core.py

```diff
@@ -1167,6 +1180,9 @@ class CosmologyCalculator(Cosmology):
             corresponding to the "HALOFIT" transformation of
             Takahashi et al. 2012 (arXiv:1208.2701).
     """
+    __eq_attrs__ = ("_params_init_kwargs", "_config_init_kwargs",
+                    "_accuracy_params", "_pk_lin", "_pk_nl",)
+
```

Comment:
Don't we need to check background and growth as well?

Reply:
To make your life easier, I'd be fine if you want to store the corresponding arrays at the python level for the comparison.

Reply:
Do we? Those arrays would uniquely describe the power spectra, right? So that extra check would be superfluous, but correct me if I'm wrong.

Reply:
Right, good call on including background and growth then.

This was way trickier than I thought it would be. If there are attributes that nest things which have to be compared, the previous implementation wouldn't work. It was watertight for simple int, float, str, and ndarray, but not for all sequences like list, tuple, str, set etc.

I created a helper function, taking advantage of the power of numpy, which can compare just about everything:
```python
def is_equal(this, other):
    """Powerful helper for equivalence checking."""
    try:
        np.testing.assert_equal(this, other)
        return True
    except AssertionError:
        return False
```

So now the lines in `__eq__` change to
```python
            for attr in self.__eq_attrs__:
                if not is_equal(getattr(self, attr), getattr(other, attr)):
                    return False
```

Reply:
How does `assert_equal` deal with complex CCL-like objects?

Reply:
It uses their internal `__eq__` method, however it is important to note that this never happens here, because what we compare is not CCLObjects, but merely arrays, sequences, numbers etc. But even if it were to compare CCLObjects, it would work.

Reply:
Hmm, aren't the attributes being compared here sometimes `MassDef`s or `Concentration`s? 
The numpy documentation does say that they must be "two objects (scalars, lists, tuples, dictionaries or numpy arrays),"

Reply:
Yes. Sorry. I answered a whole different thing. The takeaway is that numpy can handle this:
```python
>>> ccl.base.is_equal(*[ccl.CosmologyVanillaLCDM for _ in range(2)])
True
```
but
```python
>>> Mvir = ccl.halos.MassDefVir()
>>> M200m = ccl.halos.MassDef200m()
>>> ccl.base.is_equal(Mvir, M200m)
False
```